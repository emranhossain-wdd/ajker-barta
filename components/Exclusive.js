import { formatDistanceToNow } from 'date-fns';
import { useRouter } from 'next/router';
import React, { useState } from 'react'
import styles from '../styles/exclusive.module.css'
const Exclusive = ({ newses }) => {
  const latest = newses?.reverse()
  const router = useRouter()
  const [isLetest, setIsLetest] = useState(true)
  let count = 1
  return (
    <div data-testid='exclusiveId' className='pb-5'>
      <div className="container">
        <div className="grid lg:grid-cols-2 gap-3">
          <div>
            <button onClick={() => setIsLetest(true)} className={isLetest ? 'py-1 slim-border rounded-sm mb-3 px-4 border-blue-400 mr-2 bg-blue-500 text-white' : 'py-1 slim-border rounded-sm mb-3 px-4 bg-white border-blue-400 mr-2'}>Letest</button>
            <button onClick={() => setIsLetest(false)} className={!isLetest ? 'py-1 slim-border rounded-sm mb-3 px-4 border-blue-400 mr-2 bg-blue-500 text-white' : 'py-1 slim-border rounded-sm mb-3 px-4 bg-white border-blue-400 mr-2'}>Trending</button>
            {
              isLetest ? <div className='grid sm:grid-cols-2 gap-3'>
                {
                  latest?.slice(0, 8).map(news => <div onClick={() => router.push(`/${news.category}/${news.subCategory}/${news?._id}`)} className=' relative p-6 shadow-md rounded-r-lg border-r-4 border-red-600 cursor-pointer' key={news._id}>
                    <h1 className='leading-7 relative z-10 text-lg font-medium text-gray-800 hover:text-red-600 transition-colors duration-300 cursor-pointer'>{news?.heading}</h1>
                    <p style={{ transform: "translate(-50%, -50%)", }} className='absolute top-1/2 left-3/4 text-8xl font-bold text-blue-100 '>{count++}</p>
                  </div>)
                }
              </div> : <div className='grid sm:grid-cols-2 gap-3'>
                {
                  latest?.slice(8, 16).map(news => <div onClick={() => router.push(`/news/${news._id}`)} className=' relative p-6 shadow-md rounded-r-lg border-r-4 border-red-600 cursor-pointer' key={news._id}>
                    <h1 className='leading-7 relative z-10 text-lg font-medium text-gray-800 hover:text-red-600 transition-colors duration-300 cursor-pointer'>{news?.heading}</h1>
                    <p style={{ transform: "translate(-50%, -50%)", }} className='absolute top-1/2 left-3/4 text-8xl font-bold text-blue-100 '>{count++}</p>
                  </div>)
                }
              </div>
            }

          </div>
          <div>
            <h1 className={styles.exclusive}>Satireday</h1>
            <div className='grid sm:grid-cols-2 gap-3'>
              {
                latest?.slice(16, 20).map(news => <div className='cursor-pointer imageEffect' onClick={() => router.push(`/${news.category}/${news.subCategory}/${news?._id}`)} key={news?._id}>
                  <div className='overflow-hidden'>
                    <img className='w-full h-32 object-cover' src={news?.images.img1} alt="" />
                  </div>
                  <h1 className='text-lg my-1 font-medium leading-5 hover:text-red-600 transition-colors duration-300 cursor-pointer'>{news?.heading}</h1>
                  <p className='text-sm leading-4'>{news?.description[0]?.slice(0, 100)}</p>
                  <p className="px-2 mt-2 py-1 font-medium text-sm rounded-full bg-gray-100 w-fit text-blue-500">{`${formatDistanceToNow(new Date(news.publishedDate))} ago`}</p>
                </div>)
              }
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Exclusive;