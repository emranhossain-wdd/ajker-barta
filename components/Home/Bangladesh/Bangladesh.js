import { formatDistanceToNow } from "date-fns";
import Link from "next/link";
import { useRouter } from "next/router";
import React from "react";
import { IoIosArrowForward } from "react-icons/io";
import styles from '../../../styles/Bangladesh.module.css'
const Bangladesh = ({ bdnews }) => {
  const latest = bdnews.reverse()
  const col1 = latest[0];
  const col2 = latest[1];
  const col3 = latest.slice(2, 7);
  const router = useRouter()
  return (
    <div className="container h-auto mb-4">
      <div className='flex mb-4 items-center'>
        <h1 onClick={() => router.push('bangladesh')} className=' text-xl cursor-pointer font-medium text-blue-900'>Bangladesh </h1><IoIosArrowForward className='text-red-500' />
      </div>
      <div className={styles.bdRow}>
        {
          bdnews?.slice(1, 8).map(news => <div onClick={() => router.push(`/${news.category}/${news.subCategory}/${news?._id}`)} className={styles.newsBox} key={news._id}>
            <img className="object-cover rounded-sm w-32 h-24" src={news.images.img1} alt="" />
            <div>
              <h1 className="font-semibold text-black leading-5">{news.heading}</h1>
              <h2 className="hidden text-sm">{news.description[0]?.slice(0, 100)}</h2>
              <p className="px-2 mt-2 py-1 font-medium text-sm rounded-full bg-gray-100 w-fit">{`${formatDistanceToNow(new Date(news.publishedDate))} ago`}</p>
            </div>
          </div>)
        }
      </div>
    </div>
  );
};

export default Bangladesh;
