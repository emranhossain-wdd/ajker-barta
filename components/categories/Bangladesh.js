import React, { useEffect, useState } from 'react';
import styles from '../../styles/Bangladesh.module.css'
const Bangladesh = () => {
    const [news, setNews] = useState([])
    useEffect(() => {
        fetch('/Bangladesh.json')
        .then(res => res.json())
        .then(data => setNews(data))
    }, [])
    return (
        <div className='h-auto mb-4'>
           
            <h2>Bangladesh</h2>
            <div className={`grid first:row-span-3  gap-2 md:grid-cols-3 sm:grid-cols-1 ${styles.bangladesh}`}>
                {
                    news.map(item => {
                        return (<div className='flex first:row-span-3 first:flex-col-reverse justify-between auto-cols-fr' key={item.id}>
                            <div className='mr-1 w-50 first:w-full'>
                                <h1 className='font-bold'>{item.title}</h1>
                                <p>{item.date}</p>
                            </div>
                            <div className=''>
                            <img className='w-44 h-20 first:w-full first:items-start' src={item.img} alt={item.title} />
                            </div>
                        </div>)
                    }
                    )
                }
            </div>
        </div>
    );
};

export default Bangladesh;