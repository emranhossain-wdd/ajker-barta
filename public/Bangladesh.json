[
    {
      "id": "1",
      "title": "Golden era of music ended: Bollywood pays tribute to Lata Mangeshkar",
      "category": "Entertainment",
      "img": "https://images.prothomalo.com/prothomalo-english%2F2022-02%2F7aa0643f-0d30-4dd1-a868-75b4d8376a86%2F1644287211_unb_newswire.jpg?auto=format%2Ccompress&format=webp&dpr=1.0&q=70&w=640",
      "description": "From Anil Kapoor to Akshay Kumar, several members from the film industry paid tribute to legendary singer Lata Mangeshkar, who passed away on Sunday morning at the age of 92",
      "date": " 06 Feb 2022"
    },
    {
      "id": "2",
      "title": "Messi, Mbappe on target as PSG thrash sorry Lille",
      "category": "Sport",
      "img": "https://images.prothomalo.com/prothomalo-english/2021-03/82dea110-6c27-4a0c-aea7-29cd37da9fa5/prothomalo_bangla_2021_01_3bd896c8_dc7f_403a_8732_6507dc320362_IMG_9920.jpg?format=webp&dpr=1.0&q=70&w=800",
      "description": "Lionel Messi and Kylian Mbappe both scored as Paris Saint-Germain stretched their mammoth lead at the top of Ligue 1 to 13 points with a 5-1 thrashing of defending champions Lille on Sunday",
      "date": " 06 Feb 2022"
    },
    {
      "id": "3",
      "title": "Construction companies, raw material producers to meet every three months",
      "category": "Business",
      "img": "https://images.prothomalo.com/prothomalo-english%2F2022-01%2Fdc22f357-084c-450b-9c78-ac29f26f3312%2Fprothomalo_bangla_2021_04_caec4a2e_dd5e_46f5_acdd_33b8aec00561_4.jpg?auto=format%2Ccompress&format=webp&dpr=1.0&q=70&w=640",
      "description": "In this context, Bangladesh Steel Manufacturers Association (BSMA) and Bangladesh Association of Construction Industry (BACI) held a meeting at the FBCCI office recently, said a press release on Saturday",
      "date": " 06 Feb 2022"
    },
    {
      "id": "4",
      "title": "Golden era of music ended: Bollywood pays tribute to Lata Mangeshkar",
      "category": "Entertainment",
      "img": "https://images.prothomalo.com/prothomalo-english%2F2020-07%2F556be6e3-348e-4de6-8ca3-b4d1587f1854%2F1.webp?auto=format%2Ccompress&format=webp&dpr=1.0&q=70&w=640",
      "description": "From Anil Kapoor to Akshay Kumar, several members from the film industry paid tribute to legendary singer Lata Mangeshkar, who passed away on Sunday morning at the age of 92",
      "date": " 06 Feb 2022"
    },
    {
      "id": "5",
      "title": "Messi, Mbappe on target as PSG thrash sorry Lille",
      "category": "Sport",
      "img": "https://i.postimg.cc/Sx32hcvJ/sport.jpg",
      "description": "Lionel Messi and Kylian Mbappe both scored as Paris Saint-Germain stretched their mammoth lead at the top of Ligue 1 to 13 points with a 5-1 thrashing of defending champions Lille on Sunday",
      "date": " 06 Feb 2022"
    },
    {
      "id": "6",
      "title": "Construction companies, raw material producers to meet every three months",
      "category": "Business",
      "img": "https://i.postimg.cc/QdJ861Z2/business.jpg",
      "description": "In this context, Bangladesh Steel Manufacturers Association (BSMA) and Bangladesh Association of Construction Industry (BACI) held a meeting at the FBCCI office recently, said a press release on Saturday",
      "date": " 06 Feb 2022"
    },
    {
      "id": "7",
      "title": "Construction companies, raw material producers to meet every three months",
      "category": "Business",
      "img": "https://i.postimg.cc/QdJ861Z2/business.jpg",
      "description": "In this context, Bangladesh Steel Manufacturers Association (BSMA) and Bangladesh Association of Construction Industry (BACI) held a meeting at the FBCCI office recently, said a press release on Saturday",
      "date": " 06 Feb 2022"
    }
  ]