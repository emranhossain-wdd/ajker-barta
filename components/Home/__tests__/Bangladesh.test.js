import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Bangladesh from '../Bangladesh/Bangladesh'
import { render } from '@testing-library/react'
import { testArr } from '../../__tests__/Exclusive.test'

it('render Bangladesh component without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Bangladesh />, div)
})

it('testing Bangladesh with props', () => {
  const { getByTestId } = render(<Bangladesh testArr={testArr} />)
  expect(getByTestId('bangladeshId')).toBeTruthy()
})
it('maching snapshot', () => {
  const tree = renderer.create(<Bangladesh />)
  expect(tree).toMatchSnapshot()
})