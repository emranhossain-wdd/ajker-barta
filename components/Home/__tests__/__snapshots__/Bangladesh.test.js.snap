// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maching snapshot 1`] = `
<div
  className="pb-5"
  data-testid="exclusiveId"
>
  <div
    className="container"
  >
    <div
      className="grid lg:grid-cols-2 gap-3"
    >
      <div>
        <button
          className="py-1 slim-border rounded-sm mb-3 px-4 border-blue-400 mr-2 bg-blue-500 text-white"
          onClick={[Function]}
        >
          Letest
        </button>
        <button
          className="py-1 slim-border rounded-sm mb-3 px-4 bg-white border-blue-400 mr-2"
          onClick={[Function]}
        >
          Trending
        </button>
        <div
          className="grid sm:grid-cols-2 gap-3"
        />
      </div>
      <div>
        <h1
          className="exclusive"
        >
          Satireday
        </h1>
        <div
          className="grid sm:grid-cols-2 gap-3"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`maching snapshot 2`] = `
<div
  className="container h-auto mb-4"
  data-testid="bangladeshId"
>
  <div
    className="flex mb-4 items-center"
  >
    <h1
      className=" text-xl cursor-pointer font-medium text-blue-900"
      onClick={[Function]}
    >
      Bangladesh 
    </h1>
    <svg
      className="text-red-500"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
      />
    </svg>
  </div>
  <div
    className="bdRow"
  />
</div>
`;
